File: salimmo-parent/property-service/src/main/java/org/propertyservice/entities/PropertyLocation.java
package org.propertyservice.entities;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class PropertyLocation extends BaseEntity implements Serializable {
    @OneToOne(mappedBy = "propertyLocation")
    private Property property;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "city_id")
    private City city;
    private int houseNumber;
    private String street;
    private String latitude;
    private String longitude;
}

File: salimmo-frontend/src/app/models/propertylocation/propertylocation.ts
import {Property} from "../property/property";
import {City} from "../city/city";

export class Propertylocation {
  id!:number;
  property!:Property;
  city!:City;
  houseNumber!:number;
  street!:string;
  latitude!:string;
  longitude!:string;
}
